### PR TITLE
HRINT-1536 - database toggle in a single command

### PR DESCRIPTION
**Pull request by grisha-kotler**

### CODE REVIEW

File: src/Raven.Studio/typescript/common/shell/databasesManager.ts [hunk 1]
```diff
@@ -178,29 +178,36 @@ class databasesManager {
...
-                    
-                    if (!relevant) {
-                        this.onNoLongerRelevant(db);
-                    }
-                });
-
-        } else if (event.ChangeType === "Put") {
-            this.updateDatabaseInfo(db, event.DatabaseName);
+
+        switch (event.ChangeType) {
```
Comment (ml054) [inline src/Raven.Studio/typescript/common/shell/databasesManager.ts:182]: I'm not sure if this is importatnt, but previous code was checking if database is not null. Currently check is missing.

Comment (grisha-kotler) [reply src/Raven.Studio/typescript/common/shell/databasesManager.ts:182]: done

**Review by ml054 — APPROVED** [review]
studio part looks good

**Review by ayende — APPROVED** [review]

File: src/Raven.Server/ServerWide/Commands/ToggleDatabasesStateCommand.cs [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+            return Value.ToJson();
+        }
+
+        public override BlittableJsonReaderObject GetUpdatedValue(JsonOperationContext context, BlittableJsonReaderObject previousValue, long index)
+        {
+            return null;
+        }
+    }
+
+    public class ToggleParameters : IDynamicJson
```
Comment (ppekrol) [inline src/Raven.Server/ServerWide/Commands/ToggleDatabasesStateCommand.cs]: Move it inside `ToggleDatabasesStateCommand` and rename to `Parameters`

File: src/Raven.Server/ServerWide/Commands/ToggleDatabasesStateCommand.cs [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+            return new DynamicJsonValue
+            {
+                [nameof(ToggleType)] = ToggleType,
+                [nameof(DatabaseNames)] = TypeConverter.ToBlittableSupportedType(DatabaseNames),
+                [nameof(State)] = State
+            };
+        }
+    }
+
+    public enum ToggleType
```
Comment (ppekrol) [inline src/Raven.Server/ServerWide/Commands/ToggleDatabasesStateCommand.cs]: Move it inside `ToggleParameters`

File: src/Raven.Server/ServerWide/Commands/ToggleDatabasesStateCommand.cs [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+
+        public override BlittableJsonReaderObject GetUpdatedValue(JsonOperationContext context, BlittableJsonReaderObject previousValue, long index)
+        {
+            return null;
+        }
+    }
+
+    public class ToggleParameters : IDynamicJson
+    {
+        public ToggleType ToggleType { get; set; }
```
Comment (ppekrol) [inline src/Raven.Server/ServerWide/Commands/ToggleDatabasesStateCommand.cs]: rename property name to `Type`

File: src/Raven.Server/ServerWide/Commands/ToggleDatabasesStateCommand.cs [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+
+        public string[] DatabaseNames { get; set; }
+
+        public bool State { get; set; }
```
Comment (ppekrol) [inline src/Raven.Server/ServerWide/Commands/ToggleDatabasesStateCommand.cs]: I do not like that name, but I cannot think of a better one

File: src/Raven.Server/Web/System/AdminDatabasesHandler.cs [hunk 6]
```diff
@@ -837,38 +837,40 @@ public async Task EnableDatabases()
             await ToggleDisableDatabases(disable: false);
         }
 
+        [RavenAction("/admin/databases/toggle-indexing", "POST", AuthorizationStatus.Operator)]
```
Comment (ppekrol) [inline src/Raven.Server/Web/System/AdminDatabasesHandler.cs]: I do not like the inconsistency between this endpoint and `/admin/databases/dynamic-node-distribution` in naming

File: src/Raven.Server/Web/System/AdminDatabasesHandler.cs [hunk 7]
```diff
@@ -837,38 +837,40 @@ public async Task EnableDatabases()
...
+        public async Task ToggleIndexing()
+        {
+            using (ServerStore.ContextPool.AllocateOperationContext(out TransactionOperationContext context))
+            {
+                var raftRequestId = GetRaftRequestIdFromQuery();
+                var disable = GetBoolValueQueryString("disable") ?? false;
```
Comment (ppekrol) [inline src/Raven.Server/Web/System/AdminDatabasesHandler.cs]: The `/admin/databases/dynamic-node-distribution` has `enable`. Let's stick to one?

Comment (ppekrol) [reply src/Raven.Server/Web/System/AdminDatabasesHandler.cs]: I would prefer disable everywhere, we probably can change the other endpoint, it is not exposed in any client api

File: src/Raven.Server/ServerWide/Commands/ToggleDatabasesStateCommand.cs (hunk 2 above)
Comment (grisha-kotler) [reply src/Raven.Server/ServerWide/Commands/ToggleDatabasesStateCommand.cs]: done

File: src/Raven.Server/ServerWide/Commands/ToggleDatabasesStateCommand.cs (hunk 3 above)
Comment (grisha-kotler) [reply src/Raven.Server/ServerWide/Commands/ToggleDatabasesStateCommand.cs]: done

File: src/Raven.Server/ServerWide/Commands/ToggleDatabasesStateCommand.cs (hunk 4 above)
Comment (grisha-kotler) [reply src/Raven.Server/ServerWide/Commands/ToggleDatabasesStateCommand.cs]: done

File: src/Raven.Server/Web/System/AdminDatabasesHandler.cs (hunk 6 above)
Comment (grisha-kotler) [reply src/Raven.Server/Web/System/AdminDatabasesHandler.cs]: done

File: src/Raven.Server/Web/System/AdminDatabasesHandler.cs (hunk 7 above)
Comment (grisha-kotler) [reply src/Raven.Server/Web/System/AdminDatabasesHandler.cs]: We have `SetDatabaseDynamicDistributionOperation` which uses the `/admin/databases/dynamic-node-distribution` endpoint.
Changed to `enable`

File: src/Raven.Server/ServerWide/Commands/ToggleDatabasesStateCommand.cs (hunk 5 above)
Comment (grisha-kotler) [reply src/Raven.Server/ServerWide/Commands/ToggleDatabasesStateCommand.cs]: Changed to `Disable` to be more clear

**Review by ppekrol — APPROVED** [review]

**Pull request merged**